eng/common/pipelines/templates/steps/verify-resource-ref.yml

```diff
@@ -0,0 +1,32 @@
+
+steps:
+  - pwsh: |
+      Install-Module -Name powershell-yaml -RequiredVersion 0.4.1 -Force -Scope CurrentUser
+      $remoteHEAD = git rev-parse origin/master
+      $localHEAD = git rev-parse HEAD
+      $ymlfilesChanged = (git diff --name-only $remoteHEAD...$localHEAD) | Where-Object {$_ -like '*.yml'}
+      
+      foreach ($file in $ymlfilesChanged)
```

Comment:
This is enough code that we should move it out of the yml template and into a script.